test/chall-test-srv-client/client.go

```diff
@@ -0,0 +1,517 @@
+package challtestsrvclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// Client is an HTTP client for https://github.com/letsencrypt/challtestsrv's
+// management interface (test/chall-test-srv).
+type Client struct {
+	baseURL string
+}
+
+// NewClient creates a new Client using the provided baseURL, or defaults to
+// http://10.77.77.77:8055 if none is provided.
+func NewClient(baseURL string) *Client {
+	if baseURL == "" {
+		baseURL = "http://10.77.77.77:8055"
+	}
+	return &Client{baseURL: baseURL}
+}
+
+var paths = map[string]string{
+	"set-ipv4":         "/set-default-ipv4",
+	"set-ipv6":         "/set-default-ipv6",
```

Comment:
I was surprised to see this map from arbitrary string to arbitrary string since it doesn't seem to add much type safety. Indeed, the original code has a map like this too! I'm not sure exactly what purpose it serves, but we can do better here: how about making this a set of named consts? That way if we typo one we'll find out.

Alternately we could go in the direction of "do less" and just put the path names (e.g. "/set-default-ipv4") directly in the functions that use them. then there's only one named entity, the functions that get called.

Reply:
Good call.